aixcalibuha/sensitivity_analysis/sensitivity_analyzer.py

```diff
@@ -278,22 +614,547 @@ def select_by_threshold(calibration_classes, result, threshold):
+
+        for tsteps in list_tsteps:
+            tsteps_sim_results = self._load_tsteps_df(tsteps=tsteps, _filepaths=_filepaths)
+            self.logger.info(f"Analyzing these time steps.")
+            for t in tsteps:
+                result_df_tstep = self._analyze_tstep_df(time_step=t,
+                                                         tsteps_sim_results=tsteps_sim_results,
+                                                         variables=variables,
+                                                         samples=samples,
+                                                         cal_class=cal_class)
+                sen_time_dependent_list.append(result_df_tstep)
+        return sen_time_dependent_list, time_index
+
+    def _conv_global_result(self, result: dict, cal_class: CalibrationClass, analysis_variable: str):
+        glo_res_dict = self._get_res_dict(result=result, cal_class=cal_class, analysis_variable=analysis_variable)
         return pd.DataFrame(glo_res_dict, index=['global'])
 
-    def _conv_local_results(self, results: list, local_classes: list):
-        _conv_results = [self._get_res_dict(result=result, cal_class=local_class)
-                         for result, local_class in zip(results, local_classes)]
-        df = pd.DataFrame(_conv_results)
-        df.index = [c.name for c in local_classes]
+    def _conv_local_results(self, results: list, local_classes: list, verbose=False):
+        _conv_results = []
+        tuples = []
+        for class_results, local_class in zip(results, local_classes):
+            for goal, goal_results in class_results.items():
+                for av in self.analysis_variables:
+                    _conv_results.append(self._get_res_dict(result=goal_results,
+                                                            cal_class=local_class,
+                                                            analysis_variable=av))
+                    tuples.append((local_class.name, goal, av))
+        index = pd.MultiIndex.from_tuples(tuples=tuples, names=['Class', 'Goal', 'Analysis variable'])
+        df = pd.DataFrame(_conv_results, index=index)
         return df
 
     @abc.abstractmethod
-    def _get_res_dict(self, result: dict, cal_class: CalibrationClass):
+    def _get_res_dict(self, result: dict, cal_class: CalibrationClass, analysis_variable: str):
         """
         Convert the result object to a dict with the key
         being the variable name and the value being the result
-        associated to self.analysis_variable.
+        associated to analysis_variable.
         """
         raise NotImplementedError
+
+    @staticmethod
+    def plot_single(result: pd.DataFrame, **kwargs):
+        """
+        Plot sensitivity results of first and total order analysis variables.
+        For each calibration class one figure is created, which shows for each goal an axis
```

Comment:
I think for sake of structure and maintanence, all static plot methods could go into a module `plotting.py` in the folder `sensivity_analysis`. This way, this file is less long and the staticmethod is not necessary. What do you think?

Reply:
I also thought about it and I now restructured it into one module. Can you check if the imports in the __init__.py files are OK the way I did it?